arch/x86/pagetables.c

```diff
@@ -420,6 +421,18 @@ void *vmap_4k(cr3_t *cr3_ptr, void *va, mfn_t mfn, unsigned long l1_flags,
     return va;
 }
 
+void *vmap_pmm_refill_nolock(void *va, mfn_t mfn, unsigned long l1_flags) {
+    unsigned long _va = _ul(va) & PAGE_ORDER_TO_MASK(PAGE_ORDER_4K);
+
+    dprintk("%s: va: 0x%p mfn: 0x%lx\n", __func__, va, mfn);
+
+    BUG_ON(!vmap_lock);
+    va = _vmap_4k(&cr3, _ptr(_va), mfn, l1_flags, false);
```

Comment:
nit: please add a comment about directly referenced CR3 variable (this will be subject of a rework when we finally get to the better handling of pagetables)

Reply:
I change the function to match `vmap_kern_4k`/`vmap_kern_4k` with `vmap_4k_nolock`/`vmap_kern_4k_nolock` such that I reference cr3 like the other paging functions